Maximum_Product_Subset_Array/Maximum_Product_Subarray.c

```diff
@@ -0,0 +1,68 @@
+/*
+    MAXIMUM PRODUCT SUBSET OF AN ARRAY
+
+    Given an array of n elements. We need to find the maximum product possible
+    with any subset of that array.
+*/
+
+#include <stdio.h>
+#include <math.h>
+
+int neg = 0, pos = 0, zero = 0, prod = 1, max_neg = -10000000;
+
+/*
+    Approach : Two negative numbers multiply together to form a positive number.
+               Hence, the maximum product in case of even frequency of negative 
+               numbers will be the product of entire array except zero.
+               In case of odd frequency of negative numbers, we exclude the 
+               smallest negative number from the product of the array to get our
+               answer.
+               Also, in case of single negative number and no positive number in 
+               the array, the answer would be 0, if even a single 0 is present, 
+               else it would be the negative number itself.
+*/
+int solve(int temp[], int num) {
+    if (num == 1) {
+        return temp[0];
+    }
+    if (zero == num || (neg == 1 && neg + zero == num && zero)) {
+        return 0;
+    }
+    if (neg % 2 == 0) {
+        return prod;
+    }
+    return prod / max_neg;
+}
+
+int main(){
+    int num = 5;
```

Comment:
I think there is an error in the code 
If you take num=2 and temp array [0,10]. Maximum Product Subset should be 0
but the code output is 10.